packages/back-end/src/controllers/experimentLaunchChecklist.ts

```diff
@@ -83,7 +84,52 @@ export async function getExperimentCheckListByOrg(
   });
 }
 
-//TODO: Add getExperimentCheckListByProject method
+// This fetches a checklist for a specific experiment, taking into account the experiment's project
+export async function getExperimentCheckListForExperiment(
+  req: AuthRequest<null, { id: string }>,
+  res: Response,
+) {
+  const context = getContextFromReq(req);
+  const { org } = context;
+
+  const experiment = await getExperimentById(context, req.params.id);
+
+  if (!experiment) {
+    return res.status(404).json({
+      status: 404,
+      message: "Experiment not found",
+    });
+  }
+
+  const projectId = experiment.project || "";
+
+  if (!orgHasPremiumFeature(org, "custom-launch-checklist")) {
+    return res.status(200).json({
+      status: 200,
+      checklist: [],
+    });
+  }
+
+  const checklist = await getExperimentLaunchChecklist(org.id, "");
+
+  const filteredTasks: ChecklistTask[] = [];
+
+  checklist?.tasks.forEach((task) => {
+    if (!task.projects?.length) {
+      filteredTasks.push(task);
+      return;
+    }
+    if (task.projects.includes(projectId)) {
+      filteredTasks.push(task);
+      return;
+    }
+  });
```

Comment:
Nit: this can be simplified
```suggestion
  const filteredTasks = (checklist?.tasks ?? []).filter((task) => !task.projects?.length || task.projects.includes(projectId));
```